lib/PasswordValidator.php

```diff
@@ -155,4 +164,42 @@ protected function checkCommonPasswords($password) {
 		}
 	}
 
+	/**
+	 * Check if a password is in the list of breached passwords from
+	 * haveibeenpwned.com
+	 *
+	 * @param string $password
+	 * @throws HintException
+	 */
+	protected function checkHaveIBeenPwned($password) {
+		if ($this->config->getEnforceHaveIBeenPwned()) {
+			$hash = sha1($password);
+			$range = substr($hash, 0, 5);
+			$needle = strtoupper(substr($hash, 5));
+
+			$client = $this->clientService->newClient();
+
+			try {
+				$response = $client->get(
+					'https://api.pwnedpasswords.com/range/' . $range,
+					[
+						'timeout' => 5
+					]
+				);
+			} catch (\Exception $e) {
+				return;
+			}
+
+			$result = $response->getBody();
+
+			if (strpos($result, $needle)) {
+				$message = 'Password is present in compromised password list. Please chose a different password.';
+				$message_t = $this->l->t(
+					'Password is present in compromised password list. Please chose a different password.'
```

Comment:
Spotted a typo here: 
"Please chose" should be "Please choose" (lines 196 & 198)